protocol/communities_messenger_test.go

```diff
@@ -484,6 +484,74 @@ func (s *MessengerCommunitiesSuite) TestPostToCommunityChat() {
 	s.Require().True(found)
 }
 
+func (s *MessengerCommunitiesSuite) TestPinMessageInCommunityChat() {
+	ctx := context.Background()
+
+	// Create a community
+	description := &requests.CreateCommunity{
+		Membership:                  protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		Name:                        "status",
+		Color:                       "#ffffff",
+		Description:                 "status community description",
+		PinMessageAllMembersEnabled: true,
+	}
+
+	response, err := s.owner.CreateCommunity(description, true)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+
+	community := response.Communities()[0]
+	s.Require().NotNil(community)
+	s.Require().Equal(community.AllowsAllMembersToPinMessage(), true)
+
+	// Create a community chat
+	orgChat := &protobuf.CommunityChat{
+		Permissions: &protobuf.CommunityPermissions{
+			Access: protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		},
+		Identity: &protobuf.ChatIdentity{
+			DisplayName: "status-core",
+			Emoji:       "😎",
+			Description: "status-core community chat",
+		},
+	}
+	response, err = s.owner.CreateCommunityChat(community.ID(), orgChat)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+	s.Require().Len(response.Chats(), 1)
+	chat := response.Chats()[0]
+	s.Require().NotNil(chat)
+
+	s.advertiseCommunityTo(community, s.bob)
+	s.joinCommunity(community, s.bob)
+
+	inputMessage := common.NewMessage()
+	inputMessage.ChatId = chat.ID
+	inputMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+	inputMessage.Text = "message to be pinned"
+
+	sendResponse, err := s.bob.SendChatMessage(ctx, inputMessage)
+	s.Require().NoError(err)
+	s.Require().Len(sendResponse.Messages(), 1)
+
+	// bob should be able to pin the message
+	pinMessage := common.NewPinMessage()
+	pinMessage.ChatId = chat.ID
+	pinMessage.MessageId = inputMessage.ID
+	pinMessage.Pinned = true
+	sendResponse, err = s.bob.SendPinMessage(ctx, pinMessage)
+	s.Require().NoError(err)
+	s.Require().Len(sendResponse.PinMessages(), 1)
+
+	// alice did not join the community, so she should not be able to send the pin message
+	s.advertiseCommunityTo(community, s.alice)
+	sendResponse, err = s.alice.SendPinMessage(ctx, pinMessage)
```

Comment:
This needs Alice to Spectate the community too, so she has access to the Chat.

Also, I recommend creating a new `pinMessage`, as sending the same message with the same ID could cause a clash.

Finally, you can validate that the error message is the one you expect at the end, something like "user can't post".

Reply:
Done. With the latest change, this should fail in the CI until #4064 counterpart is merged to develop.